triton/stream.py

```diff
@@ -434,31 +653,40 @@ def get_stream(stream_name, config):
     if not s_config:
         raise errors.StreamNotConfiguredError()
 
-    conn = connect_to_region(s_config.get('region', 'us-east-1'))
+    if isinstance(s_config, list):
+        streams = [get_stream_by_provider(stream_name, entry) for entry in s_config]
+        return CompositeStream(streams)
+    else:
+        return get_stream_by_provider(stream_name, s_config)
+
+
+def get_stream_by_provider(stream_name, stream_config):
+    provider = stream_config.pop('provider', config.PROVIDER_AWS)
```

Comment:
This is going to remove the `provider` key from the config dictionary for all time. If this is called again, it won't be present and will default to AWS -- a bad thing if it was originally gcp.

This is easily fixed by either passing a copy of the config into the function or just copying it when it arrives to catch "all" the cases.

Reply:
Elected to copy the stream_config within this function.  Fixed.